Data_Structures_Answers.md

```diff
@@ -1,18 +1,28 @@
 Add your answers to the questions below.
+O(1) describes an algorithm that will always execute in the same time (or space) regardless of the size of the input data set.
+O(N) describes an algorithm whose performance will grow linearly and in direct proportion to the size of the input data set.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
+O(1) using the array data structure
 
 2. What is the space complexity of your ring buffer's `append` function?
+and for the space it will be O(n) since we inserting in an array
 
 3. What is the runtime complexity of your ring buffer's `get` method?
+ O(n)
 
 4. What is the space complexity of your ring buffer's `get` method?
-
+O(n)
 
 5. What is the runtime complexity of the provided code in `names.py`?
+O(n)
 
 6. What is the space complexity of the provided code in `names.py`?
+O(n)
 
 7. What is the runtime complexity of your optimized code in `names.py`?
+O(n log n) merging 2 sets then comparing them
 
 8. What is the space complexity of your optimized code in `names.py`?
+O(n) when the best case for a merge is O(n log n) then for the space it would be O(n)
```

Comment:
A working implementation that maximizes space efficiency could even exhibit O(1) space where no additional memory as been allocated.